### PR TITLE
fix: reserved keywords are checked before creating dir with forc new

### LINKED ISSUE #3565: Executing `forc new` with reserved keyword still creates the folder
Raised by @deekerno in slack:

Executing `forc new test` throws an error regarding `test` being a reserved keyword but it still creates the folder `test`.

```console
~/test-index/contracts
❯ forc new test
Error: the name `test` cannot be used as a project name, it conflicts with Sway's built-in test library

~/test-index/contracts
❯ ls
drwxr-xr-x - deekerno  9 Dec 17:43 test
```

### PR DESCRIPTION
**Pull request by kayagokalp**
closes #3565.

### CODE REVIEW

File: forc/src/cli/commands/new.rs [hunk 1]
```diff
@@ -43,6 +44,20 @@ pub(crate) fn exec(command: Command) -> Result<()> {
         path,
     } = command;
 
+    match &name {
```
Comment (bingcicle) [inline forc/src/cli/commands/new.rs:47]: An alternative way of doing this is to handle it with clap's value_parser ([fuelup example](https://github.com/FuelLabs/fuelup/blob/07381e1597568d6f200d50c97e3780bab8edd4a3/src/commands/toolchain.rs#L29-L30)). However the changes look good as they are so this is just an alternative way of doing this.

**Review by bingcicle — APPROVED** [review]

File: forc/src/cli/commands/new.rs [hunk 2]
```diff
@@ -43,6 +44,20 @@ pub(crate) fn exec(command: Command) -> Result<()> {
...
+        Some(name) => validate_name(name, "project name")?,
+        None => {
+            // If there is no name specified for the project directory name will be used by default
```
Comment (bingcicle) [inline forc/src/cli/commands/new.rs:50]:
```suggestion
            // If there is no name specified for the project, the 'path' input will be used by default
```

minor docs nit here

**Review by bingcicle — CHANGES_REQUESTED** [review]

File: forc/src/cli/commands/new.rs (hunk 2 above)
Comment (kayagokalp) [reply forc/src/cli/commands/new.rs:50]: I explicitly stated directory name as the name of the created dir is what we need to check not the path (that is why we are getting the 'file_name' from the path before checking.

Comment (bingcicle) [reply forc/src/cli/commands/new.rs]: Ah, got it. I think maybe I got a bit confused myself reading it😅

**Review by bingcicle — APPROVED** [review]

**Review by deekerno — APPROVED** [review]

**Pull request merged**